### ISSUE TITLE
[Compiler] Component

### ISSUE THREAD

**Issue by LittleSound** [vuejs/core-vapor#77]
related: #4

Most of the basic APIs related to component runtime have been basically implemented or are waiting for merging. I think it's time to complete the basic implementation of the component Compiler part.

**Comment by Ubugeeei** [comment]
About runtime code for creating component instances.

Currently, it kind of feels like I'm forcibly using the render function that has been in this repository since the beginning, and the interface is not very clean.
As I mentioned in https://github.com/vuejs/core-vapor/pull/40, before implementing the compiler, it might be better to first tidy up the runtime interfaces like `createComponent`. 🤔

**Comment by LittleSound** [comment]
You are right. After I tried it, I also found this problem, and I think it's good to make some attempts. Now we know that we must separate the creation of components from the mounting of components.

**Comment by Ubugeeei** [comment]
`createComponent` and props compilation are mostly completed in https://github.com/vuejs/core-vapor/pull/151 ! 
Directives and slots, among others, are not yet supported. 
The runtime for slots just merged a moment ago.
I think it's possible to start working on these now.

related: 
  - https://github.com/vuejs/core-vapor/issues/162
  - https://github.com/vuejs/core-vapor/issues/154
